ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/PeerSimulator/BlockFetch.hs

```diff
@@ -103,24 +110,43 @@ startKeepAliveThread registry fetchClientRegistry peerId =
         atomically retry
 
 runBlockFetchClient ::
-     (Ord peer, IOLike m, MonadTime m, MonadTimer m, HasHeader blk, HasHeader (Header blk), ShowProxy blk)
-  => peer
-  -> FetchClientRegistry peer (Header blk) blk m
+     (IOLike m, MonadTime m, MonadTimer m, HasHeader blk, HasHeader (Header blk), ShowProxy blk)
+  => Tracer m (TraceEvent blk)
+  -> PeerId
+  -> StateViewTracers m
+  -> FetchClientRegistry PeerId (Header blk) blk m
   -> ControlMessageSTM m
   -> Channel m (AnyMessage (BlockFetch blk (Point blk)))
   -> m ()
-runBlockFetchClient peerId fetchClientRegistry controlMsgSTM channel = do
-    bracketFetchClient fetchClientRegistry ntnVersion isPipeliningEnabled peerId $ \clientCtx ->
-      void $ runPipelinedPeer nullTracer codecBlockFetchId channel
+runBlockFetchClient _tracer peerId StateViewTracers {svtChainSyncExceptionsTracer} fetchClientRegistry controlMsgSTM channel = do
+    bracketFetchClient fetchClientRegistry ntnVersion isPipeliningEnabled peerId $ \clientCtx -> do
+      res <- try $ runPipelinedPeer nullTracer codecBlockFetchId channel
         (blockFetchClient ntnVersion controlMsgSTM nullTracer clientCtx)
+      case res of
+        Right _  -> pure ()
+        Left exn -> do
+          traceWith svtChainSyncExceptionsTracer $ PeerSimulatorException StateView.BlockFetchClient peerId exn
+          -- REVIEW: Do we need to print something with `tracer` here, as we do in `runChainSyncClient`?
```

Comment:
Only if we find messages that are useful to print for exceptions in the BlockFetch client. Nothing I can think of now apart from the time and size limits to the blockfetch protocol (which we do not implement in the peer simulator) or termination because of termination of the chainsync client.